File: 0200-number-of-islands/0200-number-of-islands.go
func numIslands(grid [][]byte) int {
    m:=len(grid)
    n:=len(grid[0])
    visited:=make(map[[2]int]bool)
    var dfs func(i,j int) bool 
    dfs = func(i,j int) bool {
        if i<0 || i>=m || j<0 || j>=n || visited[[2]int{i,j}]==true || grid[i][j]=='0' {
            return false
        }
        visited[[2]int{i,j}]=true
        dfs(i+1, j)
        dfs(i-1, j)
        dfs(i, j-1)
        dfs(i, j+1)
        return true
    }
    count:=0;
    for i:=0;i<m;i++ {
        for j:=0;j<n;j++ {
            if !visited[[2]int{i,j}] && grid[i][j]=='1'&& dfs(i,j) {
                count++;
            }
        }
    }
    return count
}

File: 0200-number-of-islands/0200-number-of-islands.py
class Solution:
    def numIslands(self, grid: List[List[str]]) -> int:
        m=len(grid)
        n=len(grid[0])
        count=0
        vis=set()
        def dfs(r,c):
            if r>=m or r<0 or c>=n or c<0 or grid[r][c]=='0':
                return False
            if (r,c) in vis:
                return False
            vis.add((r,c))
            dfs(r+1,c)
            dfs(r,c+1)
            dfs(r-1, c)
            dfs(r, c-1)
            return True
        for i in range(m):
            for j in range(n):
                if dfs(i,j):
                    count+=1
        return count